src/issuer/issuer.service.ts

```diff
@@ -198,12 +198,29 @@ export class IssuerService {
      * Makes a call to the agent to create a credential definition
      */
     public async createCredDef(schema_id: string, tag: string, support_revocation: boolean): Promise<any> {
-        const data = {
-            schema_id,
-            tag,
-            support_revocation,
-        };
-        return await this.agentCaller.callAgent(process.env.AGENT_ID, process.env.ADMIN_API_KEY, 'POST', 'credential-definitions', null, data);
+
+        if (support_revocation) {
+
+            const registry_size = 200;
+
+            const data = {
+                schema_id,
+                tag,
+                support_revocation,
+                registry_size
+            };
+
+            return await this.agentCaller.callAgent(process.env.AGENT_ID, process.env.ADMIN_API_KEY, 'POST', 'credential-definitions', null, data);
+        } else {
```

Comment:
There's a lot of duplicated code here. There's 2 levels of simplification. First you can move the this.agentCaller.callAgent outside the if-else because that call is identical, it's only the data that changes. The next level is that I believe you can set registry_size to 0 or null if we don't support revocation (you may want to test it out) so you can just use syntax like:
registry_size: support_revocation? 200 : 0;